purchases/src/main/kotlin/com/revenuecat/purchases/common/Dispatcher.kt

```diff
@@ -56,7 +60,10 @@ internal open class Dispatcher(
         synchronized(this.executorService) {
             if (!executorService.isShutdown) {
                 val future = if (delay != Delay.NONE && executorService is ScheduledExecutorService) {
-                    val delayToApply = (delay.minDelay.inWholeMilliseconds..delay.maxDelay.inWholeMilliseconds).random()
+                    var delayToApply = (delay.minDelay.inWholeMilliseconds..delay.maxDelay.inWholeMilliseconds).random()
+                    if (runningIntegrationTests) {
```

Comment:
Is this the only place where we use `runningIntegrationTests`? If so maybe it's cleaner to have the delay as parameter of the Dispatcher constructor?

Reply:
Hmm how do you mean? There can be 3 different types of delay: `NONE`, `DEFAULT` or `LONG`, with different durations so the delay needs to be a parameter of the `enqueue` method. Not sure if you meant something else?

Reply:
nevermind, this looks good